### DISCUSSION TITLE
Theory Question: Numerical Advantage of Updating Overlapping Cells Together

### DISCUSSION THREAD

**Discussion by MaximilianHoffmann** [miniscope/minian#184] [Q&A]
In the docs you write:

> Although we obtained YrA for each unit, it is still beneficial to group together units that heavily overlap, and carry out the update process group-wise instead of independently for each unit. This way the relative numerical relationship between heavily-overlapping units are better preserved. For this purpose, we compute pairwise Jaccard index for all units, which serve as a metric of overlap between spatial footprints of units. The parameter jac_thres is the threshold of Jaccard index above which units will be grouped together transitively.

and indeed this is, what Caiman is doing as well. I don't quite understand the reason for this. As I understand it Yra is never updated within one temporal iteration, right?  So how do these different optimization problems depend on each other?
Thanks for any help and sorry if this issue tracker is not quite the right place for this.

**Comment by phildong** [comment]
You are correct that `YrA` is not updated at all, it serve as the input/data the model is fitting to.
When updating multiple cells together, the input change from 1d (time) to 2d (cell x time) and a single optimization is performed with this 2d matrix of multiple cells.
In theory this can lead to more accurate estimation of spikes for overlapping cells, especially when some cells are much brighter than others.

**Reply by MaximilianHoffmann** [reply]
Thanks for your answer! I am afraid, that this is what I don't get. The optimization problem only sees the Yra, the background and estimates c and s, right? But all are (cell x time) and the overlap is at least not explicitly data, that is given into the program. So what is the mathematical reason, that the estimation could be better for overlapping cells, if this information isn't actually available. Is it because of the sparsity constraints?

**Reply by phildong** [reply]
Yes! In ideal situation when the sparsity constraints are playing a strong role, it can eliminate some false spikes when the `YrA` of multiple (overlapping) cells are considered altogether. But in practice we found that this doesn't really happen much and most of the demixing are done by the computation of `YrA`, and that's why this is "multiple cells update" is disabled by default.
